### PR TITLE
Add new `require-meta-default-options` rule

### LINKED ISSUE #499: Rule suggestion: `require-meta-default-options`
https://github.com/eslint/eslint/pull/17656 added support for `meta.defaultOptions` in rules. See also https://github.com/eslint/rfcs/blob/main/designs/2023-rule-options-defaults/README.md.

A new lint rule should report:

* if `meta.schema` is defined and not just `[]` and `meta.defaultOptions` is not defined
* if `meta.schema === []` but `meta.defaultOptions` _is_ defined
* if `meta.defaultOptions` is not an array
* if `meta.defaultOptions` is an empty array (Only exception in the core ESLint repo is [the `id-denylist` rule](https://github.com/eslint/eslint/pull/17656/files#diff-4f692363a6ac7a32ec4bafcd63a3b22c43ddaebae255f2bce09c4397ad9389a6R107), so maybe we should add an option to disable this check for certain rules? Or just be okay with having to add an ESLint disable comment for such cases.)

### PR DESCRIPTION
**Pull request by FloEdelmann**
Fixes #499.

### CODE REVIEW

**Comment by bmish** [comment]
@JoshuaKGoldberg

**Review by JoshuaKGoldberg — COMMENTED** [review]
LGTM, thanks for the ping & thanks for making this!

Left some questions that I'll defer to a maintainer.

File: lib/rules/consistent-output.js
Comment (FloEdelmann) [reply lib/rules/consistent-output.js:35]: See #498 -> #503 :wink: 

I didn't remove `default` from the schemas here because eslint-doc-generator doesn't support `defaultOptions` yet (see https://github.com/bmish/eslint-doc-generator/issues/513), so removing `default`s would make the docs worse.

File: tests/lib/rules/require-meta-default-options.js
Comment (FloEdelmann) [reply tests/lib/rules/require-meta-default-options.js:57]: Some rules accept a variable length of arguments, e.g. [`vue/no-restricted-v-on`](https://eslint.vuejs.org/rules/no-restricted-v-on.html) (but actually, the default value should be `[]` here, which is also forbidden by the rule, so maybe something needs to be done anyway).

Also, some rules have `anyOf` or `oneOf` in their root schema, with different number of properties in each case, e.g. [`@stylistic/function-call-spacing`](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/rules/function-call-spacing/function-call-spacing._ts_.ts). Enforcing a strict number of default options is difficult here.

File: docs/rules/require-meta-default-options.md
Comment (FloEdelmann) [reply docs/rules/require-meta-default-options.md:1]: I agree, but that seems a bit long. Maybe just adding the small word `only` already helps?

```suggestion
# Require only rules with options to implement a `meta.defaultOptions` property (`eslint-plugin/require-meta-default-options`)
```

Comment (FloEdelmann) [reply docs/rules/require-meta-default-options.md]: Changed in [`011f8b2`](https://github.com/eslint-community/eslint-plugin-eslint-plugin/pull/502/commits/011f8b23f9c2c2a7dd0199818b50b182c87d551f).

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks! 👍

**Comment by aladdin-add** [comment]
it has been shipped in [v6.4.0](https://github.com/eslint-community/eslint-plugin-eslint-plugin/releases/tag/v6.4.0) 🎉

**Pull request merged**
